docs/contributing/code-contributions.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Code Contributions
+---
+
+The beauty of contributing to open source is that you can clone your favorite framework, get it running locally, and test out experiments and changes in real time! Way to feel like a wizard.
+
+This page includes instructions specific to the Gatsby core codebase, blog, and website. To start setting up the Gatsby repo on your machine, check out the page on [setting up your local dev environment](/contributing/setting-up-your-local-dev-environment/). For additional instructions on contributing to the docs, visit the [docs contributions page](/contributing/docs-contributions/).
+
+## Making changes to the website
+
+If you want to make more changes to the website functionality or documentation, that is, change layout components or templates, add sections/pages, follow the steps for [setting up your local dev environment](/contributing/setting-up-your-local-dev-environment/). You can then spin up your own instance of the Gatsby website and make/preview your changes before raising
+a pull request.
+
+## Making changes to the starter library
+
+Note: You don't need to follow these steps to submit to the starter library. This is only necessary if you'd like to contribute to the functionality of the starter library. To submit a starter, [follow these steps instead](/contributing/submit-to-starter-library/).
+
+To develop on the starter library, you'll need to supply a GitHub personal access token.
+
+1. Create a personal access token in your GitHub [Developer settings](https://github.com/settings/tokens).
+2. In the new token's settings, grant that token the "public_repo" scope.
+3. Create a file in the root of `www` called `.env.development`, and add the token to that file like so:
+
+```text:title=.env.development
+GITHUB_API_TOKEN=YOUR_TOKEN_HERE
+```
+
+The `.env.development` file is ignored by git. Your token should never be committed.
+
+## Contributing to the blog
+
+Note: Before adding a blog post ensure you have approval from a member of the Gatsby team. You can do this by [opening an issue](https://github.com/gatsbyjs/gatsby/issues/new/choose) or contacting [@gatsbyjs on Twitter](https://twitter.com/gatsbyjs).
+
+To add a new blog post to the gatsbyjs.org blog:
+
+- Clone [the Gatsby repo](https://github.com/gatsbyjs/gatsby/) and navigate to `/www`
+- Run `yarn` to install all of the website's dependencies.
```

Comment:
Not necessary, but maybe we link to a "Why Yarn?" section of the existing [documentation here](https://www.gatsbyjs.org/docs/how-to-contribute/#using-yarn)? It's a pretty common question as to _why_ we require it.